GFDL_tools/read_climate_nudge_data.F90

```diff
@@ -579,23 +577,35 @@ end subroutine check_axis_size
 
 !------------------------------------
 
- subroutine check_resolution (axis_len)
- integer, intent(in) :: axis_len(:)
+ subroutine check_resolution (fileobj, field_name)
 
-   if (size(axis_len(:)) .lt. 2) then
+ type(FmsNetcdfFile_t), intent(in) :: fileobj
+ character(len=*), intent(in) :: field_name
+
+ integer :: nd
+ integer, allocatable :: siz(:)
+
+   nd = get_variable_num_dimensions(fileobj, field_name)
+
+   if (nd .lt. 2) then
       call error_mesg ('read_climate_nudge_data_mod', 'incorrect number of array dimensions', FATAL)
    endif
-   if (axis_len(1) .ne. global_axis_size(INDEX_LON)) then
+
+   allocate(siz(nd))
+   call get_variable_size(fileobj, field_name, siz)
+
+   if (siz(1) .ne. global_axis_size(INDEX_LON)) then
       call error_mesg ('read_climate_nudge_data_mod', 'incorrect array dimension one', FATAL)
    endif
-   if (axis_len(2) .ne. global_axis_size(INDEX_LAT)) then
+   if (siz(2) .ne. global_axis_size(INDEX_LAT)) then
       call error_mesg ('read_climate_nudge_data_mod', 'incorrect array dimension two', FATAL)
    endif
-   if (size(axis_len(:)) .gt. 3) then
-      if (axis_len(3) .ne. global_axis_size(INDEX_LEV)) then
+   if (nd .gt. 3) then
+      if (siz(3) .ne. global_axis_size(INDEX_LEV)) then
```

Comment:
This assumes any index beyond 3 will not need to be checked.  Should the index checking be put into a loop with limits of 1,min(nd,3) or should it go until it reaches nd or finds an unlimited?

Reply:
@bensonr It looks like global_axis_size has a size of 3:
https://github.com/NOAA-GFDL/GFDL_atmos_cubed_sphere/blob/f724ade2ad8ac538c5af410387ed5e90cc7e5cc1/GFDL_tools/read_climate_nudge_data.F90#L80

So even if it the index is greater than 3, there wouldn't be anything to compare the dimensions to.